### PR TITLE
Update log.js and package.json cleanup

### PR DESCRIPTION
**Pull request by dada1134**
### Proposed Semver Increment Bump: [MAJOR/MINOR/PATCH]

Adds a feature so minor?

### Changes Proposed in this Pull Request (List new items in CHANGELOG.MD)

- add an option to disable timestamps in logs
- log now adds the timestamp to every line
- reorder package.json and add `homepage` field

### (If Applicable) What Issue does it fix?

Nothing

### CODE REVIEW

**Review by UnseenFaith — DISMISSED** [review]
Few things.
A) I don't think we need an extra module to do padding. 
B) There should be a configuration option to disable color instead of just removing it automatically.

**Comment by dada1134** [comment]
> I don't think we need an extra module to do padding.

Here's a good explanation on why I used a module @UnseenFaith: https://github.com/sindresorhus/ama/issues/10#issuecomment-117766328

File: package.json [hunk 1]
```diff
@@ -1,38 +1,40 @@
...
+  "scripts": {
+    "lint": "npm run test -- --fix",
+    "test": "eslint classes commands finalizers functions inhibitors events client.js"
   },
-  "license": "ISC",
   "dependencies": {
     "chalk": "^1.1.3",
     "discord.js": "hydrabolt/discord.js",
     "fs-nextra": "0.0.3",
+    "lpad": "^2.0.1",
```
Comment (bdistin) [inline package.json]: .padEnd() .padStart() are available native in node V8.0.0.

You do not need a package for this functionality...

**Review by bdistin — CHANGES_REQUESTED** [review]

File: inhibitors/requiredFuncs.js [hunk 2]
```diff
@@ -11,6 +11,6 @@ exports.run = (client, msg, cmd) => {
   cmd.conf.requiredFuncs.forEach((func) => {
     if (!client.funcs.hasOwnProperty(func)) funcs.push(func);
   });
-  if (funcs.length > 0) return `The client is missing **${funcs.join(", ")}**, and cannot run.`;
+  if (funcs.length > 0) return `The client is missing the **${funcs.join(", ")}** function and cannot run.`;
```
Comment (bdistin) [inline inhibitors/requiredFuncs.js]: Multiple functions can be missing.

File: events/log.js [hunk 3]
```diff
@@ -1,33 +1,36 @@
...
+          break;
+          // no default
+      }
+    }
   }
-  return error;
-}
+
+  if (type === "debug") type = "log";
+  console[type](data.split("\n").map(str => str.padStart(timestamp.length + str, timestamp)).join("\n"));
```
Comment (bdistin) [inline events/log.js]: First Arg of padStart should be a number. Adding a number to a string makes it a string. (drop ` + str` to be correct)... But doing that is pointless, you should be concatenating here ~~with template literals~~.

File: events/log.js [hunk 4]
```diff
@@ -1,33 +1,36 @@
...
-  error = error.stack || error.message || error;
+  data = data.stack || data.message || data;
+  if (typeof data === "object" && typeof data !== "string" && !Array.isArray(data)) data = require("util").inspect(data, { depth: 0, colors: true });
+  if (Array.isArray(data)) data = data.join["\n"];
 
-  if (typeof error === "object" && typeof error !== "string") {
-    return require("util").inspect(error, { depth: 0, colors: true });
+  let timestamp = "";
+  if (!client.config.disableLogTimestamps) {
+    timestamp = moment().format("YYYY-MM-DD HH:mm:ss ");
```
Comment (bdistin) [inline events/log.js]: You have dropped the [] wrapping the timestamp. Please add it back.

**Review by bdistin — CHANGES_REQUESTED** [review]

File: events/log.js (hunk 3 above)
Comment (dada1134) [reply events/log.js]: I'm not sure how you wanted to fix it but I fixed it and tested it.

Comment (bdistin) [reply events/log.js]: data.split("\n").map(str => timestamp + str).join("\n")

Comment (dada1134) [reply events/log.js]: Haha of course, that's way simpler!

**Review by bdistin — DISMISSED** [review]
👍

File: inhibitors/requiredFuncs.js [hunk 5]
```diff
@@ -11,6 +11,6 @@ exports.run = (client, msg, cmd) => {
   cmd.conf.requiredFuncs.forEach((func) => {
     if (!client.funcs.hasOwnProperty(func)) funcs.push(func);
   });
-  if (funcs.length > 0) return `The client is missing the **${funcs.join(", ")}** function and cannot run.`;
+  if (funcs.length > 0) return `The client is missing the **${funcs.join(", ")}** ${funcs.length > 1 ? "functions" : "function"} and cannot run.`;
```
Comment (kyranet) [inline inhibitors/requiredFuncs.js]: You can also do 

```js
`function${funcs.length > 1 ? "s" : ""} and cannot run.`
```

**Review by kyranet — DISMISSED** [review]
👍

**Review by CyberiumShadow — APPROVED** [review]

**Pull request merged**